flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkConfigBuilder.java

```diff
@@ -177,19 +192,21 @@ public FlinkConfigBuilder applyJobManagerSpec() throws IOException {
         return this;
     }
 
-    public FlinkConfigBuilder applyTaskManagerSpec() throws IOException {
+    protected FlinkConfigBuilder applyTaskManagerSpec() throws IOException {
         if (spec.getTaskManager() != null) {
             setResource(spec.getTaskManager().getResource(), effectiveConfig, false);
-            setPodTemplate(
-                    spec.getPodTemplate(),
-                    spec.getTaskManager().getPodTemplate(),
-                    effectiveConfig,
-                    false);
+            if (forDeployment) {
+                setPodTemplate(
```

Comment:
The temp file may still keep for a long time during the operator running, because it clean up relies on the `deleteOnExit`. Can we expire the temp files when the cache are expired ?

Reply:
I will try to add that, good idea

Reply:
I think over it again, It may bring the sync problem if the config is just used to deploy and the async replace is scheduled. Maybe the cache lifecycle can be align to the job lifecycle, which means we can cleanup the cache and config in the cleanup method ?  WDYT ?

Reply:
I have added cleanup for the temp files. Also with this I removed the `forDeployment` logic because it is not really necessary anymore (temp files will be cleaned up anyways.)

Set the expiration after access to 1 hour to avoid any problems with cleanup during deployment.

Reply:
But you are right this could still be a problem if the clean up is triggered based on the configmap update, I will think about it a little

Reply:
@Aitozi I have pushed a fix for this, instead of invalidating the cache I bump a config version long field. This will avoid race conditions between default config updates and temp file cleanup

Reply:
It looks good now, +1 for this otherwise two other minor comments.